src/tcgwars/logic/impl/gen4/GreatEncounters.groovy

```diff
@@ -469,8 +469,15 @@ public enum GreatEncounters implements LogicCardInfo {
           pokePower "Cosmic Power", {
             text "Once during your turn , you may choose up to 2 cards from your hand and put them on the bottom of your deck in any order. If you do, draw cards until you have 6 cards in you hand. This power can’t be used if Claydol is affected by a Special Condition."
             actionA {
-            }
-          }
+              checkNoSPC()
+              assert my.hand.notEmpty : "You have no cards in your hand."
```

Comment:
`checkLastTurn()` could also be added here too

Reply:
Fixed